components/src/components/atoms/MenuPlacement/MenuPlacement.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from 'react'
+import * as ReactDOM from 'react-dom'
+import { CSSProperties } from 'styled-components'
+
+const throttle = (callback: (...args: any[]) => void) => {
+  let requestId: number | null = null
+  let lastArgs: any[]
+
+  const throttled = (...args: any[]) => {
+    lastArgs = args
+    if (requestId) return
+    requestId = requestAnimationFrame(() => {
+      requestId = null
+      callback(...lastArgs)
+    })
+  }
+
+  throttled.cancel = () => {
+    if (requestId) cancelAnimationFrame(requestId)
+    requestId = null
+  }
+
+  return throttled
+}
+
+type MenuPortalProps = {
+  appendTo: HTMLElement | null
+  control: HTMLElement | null
+  listenTo?: HTMLElement | null
+}
+
+const MenuPortal: React.FC<React.PropsWithChildren<MenuPortalProps>> = ({
+  appendTo,
+  control,
+  listenTo,
+  children,
+}) => {
+  // The position and size of the control relative to the appendTo element
+  const [placement, setPlacement] = React.useState<CSSProperties>({})
+
+  // Calculates where the control element would be in the appendTo element.
+  // Throttle to limit the number of times the calculation is made.
+  const calculatePlacement = React.useMemo(
```

Comment:
I removed the throttle function since after some research, onscroll events are already tied to the refresh rate of the browser so using requestAnimationFrame actually doesn't do anything since they are on the same timing.